File: testdata/jennies/builders/panel_builders/JavaBuilder/panelbuilder/PanelBuilder.java
package panelbuilder;

import java.util.List;

public class PanelBuilder implements cog.Builder<Options> {
    protected final Options internal;
    
    public PanelBuilder() {
        this.internal = new Options();
    }
    public PanelBuilder onlyFromThisDashboard(Boolean onlyFromThisDashboard) {
        this.internal.onlyFromThisDashboard = onlyFromThisDashboard;
        return this;
    }
    
    public PanelBuilder onlyInTimeRange(Boolean onlyInTimeRange) {
        this.internal.onlyInTimeRange = onlyInTimeRange;
        return this;
    }
    
    public PanelBuilder tags(List<String> tags) {
        this.internal.tags = tags;
        return this;
    }
    
    public PanelBuilder limit(Integer limit) {
        this.internal.limit = limit;
        return this;
    }
    
    public PanelBuilder showUser(Boolean showUser) {
        this.internal.showUser = showUser;
        return this;
    }
    
    public PanelBuilder showTime(Boolean showTime) {
        this.internal.showTime = showTime;
        return this;
    }
    
    public PanelBuilder showTags(Boolean showTags) {
        this.internal.showTags = showTags;
        return this;
    }
    
    public PanelBuilder navigateToPanel(Boolean navigateToPanel) {
        this.internal.navigateToPanel = navigateToPanel;
        return this;
    }
    
    public PanelBuilder navigateBefore(String navigateBefore) {
        this.internal.navigateBefore = navigateBefore;
        return this;
    }
    
    public PanelBuilder navigateAfter(String navigateAfter) {
        this.internal.navigateAfter = navigateAfter;
        return this;
    }
    public Options build() {
        return this.internal;
    }
}

File: testdata/jennies/builders/panel_builders/CSharpBuilder/src/Grafana/Foundation/Panelbuilder/PanelBuilder.cs
namespace Grafana.Foundation.Panelbuilder;


public class PanelBuilder : Cog.IBuilder<Options>
{
    protected readonly Options @internal;

    public PanelBuilder()
    {
        this.@internal = new Options();
    }

    public PanelBuilder OnlyFromThisDashboard(bool onlyFromThisDashboard)
    {
        this.@internal.OnlyFromThisDashboard = onlyFromThisDashboard;
        return this;
    }

    public PanelBuilder OnlyInTimeRange(bool onlyInTimeRange)
    {
        this.@internal.OnlyInTimeRange = onlyInTimeRange;
        return this;
    }

    public PanelBuilder Tags(List<string> tags)
    {
        this.@internal.Tags = tags;
        return this;
    }

    public PanelBuilder Limit(uint limit)
    {
        this.@internal.Limit = limit;
        return this;
    }

    public PanelBuilder ShowUser(bool showUser)
    {
        this.@internal.ShowUser = showUser;
        return this;
    }

    public PanelBuilder ShowTime(bool showTime)
    {
        this.@internal.ShowTime = showTime;
        return this;
    }

    public PanelBuilder ShowTags(bool showTags)
    {
        this.@internal.ShowTags = showTags;
        return this;
    }

    public PanelBuilder NavigateToPanel(bool navigateToPanel)
    {
        this.@internal.NavigateToPanel = navigateToPanel;
        return this;
    }

    public PanelBuilder NavigateBefore(string navigateBefore)
    {
        this.@internal.NavigateBefore = navigateBefore;
        return this;
    }

    public PanelBuilder NavigateAfter(string navigateAfter)
    {
        this.@internal.NavigateAfter = navigateAfter;
        return this;
    }

    public Options Build()
    {
        return this.@internal;
    }
}
